VideoApp/VideoApp/TwilioVideo/Captions/CaptionsManager.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  Copyright (C) 2022 Twilio, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Combine
+import TwilioVideo
+
+/// Receive captions on a data track from the `media-transcriber` participant. Discard old captions
+/// when time has elapsed or space for new captions is needed. The `media-transcriber` participant
+/// is only for receiving captions and should not be visible in the UI.
+class CaptionsManager: NSObject, ObservableObject {
+    @Published var captions: [Caption] = []
+    private var subscriptions = Set<AnyCancellable>()
+
+    var transcriber: RemoteParticipant? {
+        didSet {
+            oldValue?.delegate = nil
+            oldValue?.remoteDataTracks.forEach { $0.remoteTrack?.delegate = nil }
+            subscriptions.removeAll()
+            
+            if let transcriber = transcriber {
+                transcriber.delegate = self
+                
+                Timer.publish(every: 1, on: .main, in: .default)
+                    .autoconnect()
+                    .sink { [weak self] date in
+                        self?.captions.removeAll(where: { date.timeIntervalSince($0.date) > 10 })
+                    }
+                    .store(in: &subscriptions)
+            }
+        }
+    }
+}
+
+extension CaptionsManager: RemoteParticipantDelegate {
+    func didSubscribeToDataTrack(
+        dataTrack: RemoteDataTrack,
+        publication: RemoteDataTrackPublication,
+        participant: RemoteParticipant
+    ) {
+        dataTrack.delegate = self
+    }
+}
+
+extension CaptionsManager: RemoteDataTrackDelegate {
+    func remoteDataTrackDidReceiveString(remoteDataTrack: RemoteDataTrack, message: String) {
+        guard
+            let data = message.data(using: .utf8),
+            let transcription = try? JSONDecoder().decode(Transcription.self, from: data),
+            let caption = Caption(transcription: transcription)
+        else {
+            return
+        }
+
+        if let index = captions.firstIndex(where: { $0.id == caption.id }) {
+            captions[index] = caption
+        } else {
+            captions.append(caption)
+            
+            if captions.count > 3 {
```

Comment:
Are we worried about the text from <=3 captions potentially filling the screen?

Reply:
Good question. Looks like it just starts truncating messages. Shouldn't happen in practice but seems like probably the ideal solution if it does happen. I should add a UI preview for this.

<img width="443" alt="Screen Shot 2022-05-24 at 4 00 34 PM" src="https://user-images.githubusercontent.com/1930363/170139412-1ae27c00-a46f-4ae6-98cc-994861cc5b15.png">

<img width="461" alt="Screen Shot 2022-05-24 at 4 08 07 PM" src="https://user-images.githubusercontent.com/1930363/170140174-6dd520b6-2ac7-4ac6-95fc-413c55e13ed3.png">